torch/package/_custom_import_pickler.py

```diff
@@ -36,9 +41,15 @@ def save_global(self, obj, name=None):
                 (obj, module_name, name)) from None
         else:
             if obj2 is not obj:
-                raise PicklingError(
-                    "Can't pickle %r: it's not the same object as %s.%s" %
-                    (obj, module_name, name))
+                # CHANGED: More specific error message in the case of mangling.
+                obj2_parent_module_name = getattr(obj2, "__module__", orig_module_name)
+                msg = f"Can't pickle {obj}: it's not the same object as {obj2_parent_module_name}.{name}."
+
+                if is_mangled(getattr(obj, "__module__", "")) or is_mangled(obj2_parent_module_name):
+                    msg += ("\nThis is likely due to a dependency collision in torch.package; "
```

Comment:
nit: maybe an explanation of how to resolve the problem 'namespace defined by package.importers resolves {untangled_name} to ... but the object is coming from ..., a module loaded from a torch.package. If you want to save this object you need to add the importer for the package to the list of importers ..."